compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -740,14 +741,16 @@ private boolean checkStructuredPatternSimilarity(BLangVariable precedingVar, BLa
 
             for (int i = 0; i < precedingRecVar.variableList.size(); i++) {
                 BLangRecordVariableKeyValue precedingKeyValue = precedingRecVar.variableList.get(i);
-                if (!recVarAsMap.containsKey(precedingKeyValue.key.value)) {
-                    return false;
+                if (recVarAsMap.containsKey(precedingKeyValue.key.value)) {
```

Comment:
I think it is better to inverse the `if` here to reduce the number of nested statements.